app/bundles/MessengerBundle/Config/parameters.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+use Mautic\MessengerBundle\MauticMessengerRoutes;
+/** @var ContainerBuilder $container */
+use Mautic\MessengerBundle\Message\EmailHitNotification;
+use Mautic\MessengerBundle\Message\PageHitNotification;
+use Mautic\MessengerBundle\Middleware\SynchronousExtrasMiddleware;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+$container->loadFromExtension('framework', [
+    'messenger' => [
+        'buses' => [
+            'messenger.bus.default' => [
+                'default_middleware'    => true,
+                'middleware'    => [
+                    SynchronousExtrasMiddleware::class,
+                ],
+            ],
+        ],
+        'failure_transport' => 'failed',
+        'routing'           => [
+            PageHitNotification::class  => MauticMessengerRoutes::SYNC,
+            EmailHitNotification::class => MauticMessengerRoutes::SYNC,
+        ],
+        'transports'        => [
+            MauticMessengerRoutes::SYNC      => 'sync://',
```

Comment:
I do not think we should use `SYNC` at all, everything in Mautic must be queued then processed

Reply:
well sync is the default. it does use messenger completely. it just does not use AMQP on the way. just configure another transport, seamlessly.
we should notforce people to use AMQP.
secondly, i have just added a functionality that was there but did not work if queue eneabled. the cookie on email hit. if you chose not to use the QueueBundle it simply went on normally and set a cookie in the process.
with QueueEnabled the client was no longer active request, so nothing to set. that is fine even though it is a drawback from mautic's standard functionality. it is worth it. we did not force them to use the queuebundle before.
my messenger did not offer cookie possibilities at all. it does now for the **sync** transport. so everything works like before and everything uses messenger in the same time.

Reply:
@galvani, please finish the first code that is ready to test for weeks, then make suggestions. Not need it now, it could be part of another PR